docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+
+This method is used to deposit minted cycles into a specific subaccount in the [cycles ledger.](/docs/concepts/tokens/cycles/cycles-ledger) It supports minting from both legacy and ICRC-1-style transfers.
+
+The CMC expects the payment to follow a strict structure that encodes the **intent** and **destination**:
+
+-   The subaccount used in the transfer must be derived from the caller principal using the standard 32-byte encoding (see “Shared Logic” section).
+-   The **ICP transfer's memo field** must explicitly indicate the intent to mint cycles. This can be:
+    -   For legacy `transfer` calls: a `u64` memo with value (in decimal):
+        ```
+        1414416717
+        ```
+    -   For ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`): a `blob` memo equal to:
+        ```
+        "\4d\49\4e\54\00\00\00\00" // ASCII "MINT" + 4 zero bytes
+        ```
+
+#### Parameters
+
+```candid
+record {
+  block_index: nat64;       // Block index of the ICP ledger payment
+  to_subaccount: opt blob;   // 32-byte subaccount to credit in the cycles ledger
+  deposit_memo: opt blob;    // Optional application-specific memo that will be used in the cycles ledger's deposit transaction
+}
+```
+
+#### Returns
+
+```candid
+variant {
+  Ok: record {
+    block_index: nat;
+    minted: nat;
+    balance: nat;
+  };
+  Err: NotifyError;
+}
+```
+
+-   **Ok**: A record that includes:
+    -   `block_index`: The cycles ledger block index of the minting transaction
+    -   `minted`: The amount of cycles created from the ICP payment
+    -   `balance`: The new balance of the target cycles ledger subaccount
+-   **Err**: Indicates why the minting failed (e.g., incorrect memo, controller mismatch, payment already processed, etc.).
+    For information about retry semantics, error handling, and caching behavior, see [Notify Methods – Shared Behavior](#notify-methods--shared-behavior-retry-error-handling-and-caching).
+
+## Notify Methods – Shared Behavior (Retry, Error Handling, and Caching)
+
+All `notify_*` methods (such as `notify_create_canister` and `notify_top_up`) follow the same core semantics:
+
+-   Before processing any specific operation (like create or top-up), the CMC checks the memo of the incoming ICP transfer identified by `block_index`. If the memo (either the legacy `u64` memo or the ICRC-1 `blob` memo interpreted as a little-endian `u64`) does **not** match one of the recognized operation memos (`1095062083` for create, `1347768404` for top-up, `1414416717` for mint), the CMC attempts to automatically refund the ICP amount (minus the standard ledger transfer fee) back to the sender's account (`from` account in the transfer). If the refund succeeds or fails terminally, the call will return a `NotifyError::Refunded` or `NotifyError::InvalidTransaction` respectively. If the refund attempt fails transiently (e.g., ledger unavailable), the status is cleared, allowing a retry.
```

Comment:
the retry (last line here) has then to be triggered manually? 
so the CMC just tries ones can then stops trying?